pkg/steps/bundle_source.go

```diff
@@ -80,7 +79,13 @@ func (s *bundleSourceStep) run(ctx context.Context) error {
 		"",
 	)
 
-	return handleBuilds(ctx, s.client, s.podClient, *build, newImageBuildOptions(s.multiArch))
+	// It doesn't matter where the bundle image will be built, but to keep the consistency
+	// let's force it to run on an amd64 node.
+	build.Spec.NodeSelector = map[string]string{"kubernetes.io/arch": "amd64"}
```

Comment:
Removing this at all would allow to schedule on arm64 whenever arm64 nodes are available and also avoid amd64 nodes scale out when arm64 resources are available, but amd64 are not.

Why don't we just set it to build on amd64? What's the 'consistency' we are trying to guarantee here?

If it's about avoiding errors we are not aware of, should we consider a story to remove this node selector at one point?

Reply:
I would prefer forcing an amd64 build as a default behaviour to keep the consistency and allow the user to override it by specifying something like `allowAnyArchitecture: true` in the configuration.

Reply:
I would refrain from exposing this specificity to the users, especially in a case like this, where it is unneeded. Users are not even meant to know about this specific topic and also struggle to get it if they don't know all the nuances of multi-arch/manifest-list images, multi-arch clusters, etc... 

We also discussed this PR and the broader issue of 'default node selectors' in CI a few minutes ago with other folks during the multiarch office hour. 

We could add this topic to the agenda of the next sync meeting on Monday. Wdyt?

cc @tvardema @jaypoulz @jeffdyoung

Reply:
For now, I removed the amd64 defaulting and just left a comment for future observations if needed.

Reply:
Thanks @droslean 

/lgtm